src/app/[lng]/(common)/(common)/mypage/page.tsx

```diff
@@ -1,56 +1,64 @@
-import { redirect } from 'next/navigation';
+'use client';
 
-import { auth } from '@/api/auth/auth';
-import { Notice } from '@/api/notice/notice';
-import { getAllNotices } from '@/api/notice/notice-server';
-import { createTranslation, PropsWithLng } from '@/app/i18next';
+import Link from 'next/link';
+import { useRouter } from 'next/navigation';
+import { signOut, useSession } from 'next-auth/react';
+import { useEffect } from 'react';
 
+import Button from '@/app/components/atoms/Button';
+import { PropsWithLng } from '@/app/i18next';
+import { useTranslation } from '@/app/i18next/client';
+
+import ChangeLanguageBox from './ChangeLanguageBox';
+import MypageBox from './MypageBox';
+import MypageButtons from './MypageButton';
 import MypageProfile from './MypageProfile';
-import MypageTable from './MypageTable';
-
-export default async function MyPage({
-  params: { lng },
-}: {
-  params: PropsWithLng;
-}) {
-  const { t } = await createTranslation(lng);
-  const userData = await auth();
-  const remindedNotice: Notice[] = (
-    await getAllNotices({ my: 'reminders', lang: lng })
-  ).list;
-  const ownNotice: Notice[] = (
-    await getAllNotices({ my: 'own', limit: 5, orderBy: 'recent', lang: lng })
-  ).list;
-
-  if (!userData) return redirect(`/${lng}/login`);
-
-  const { user } = userData;
+
+export default function MyPage({ params: { lng } }: { params: PropsWithLng }) {
+  const { t } = useTranslation(lng);
+
+  const { data: sessionData } = useSession();
```

Comment:
```suggestion
  const { data: sessionData } = useSession();

  if (!sessionData) {
    redirect('/login');
  }
```

redirect는 next/navigation에서 import 하였습니다

Reply:
밑에 sessionData?.user 도 다 없앨 수 있을겁니당

Reply:
아 이런기능이 있었네 🥲